experimental/web/index.html

```diff
@@ -0,0 +1,152 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+    <meta charset="utf-8" />
+    <title>Oak emulator</title>
+</head>
+
+<body>
+    <script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>
+    <script type="module">
+        function init() {
+            var app = new Vue({
+                el: '#app',
+                data: {
+                    // Exports of the loaded Module.
+                    exports: [],
+                    // Import of the loaded Module.
+                    imports: [],
+                    // Oak ABI calls trace.
+                    trace: [],
+                    // Loaded module.
+                    module: null,
+                    // Loaded module instance.
+                    instance: null,
+                },
+                methods: {
+                    // Read the specified file and load it as a Wasm module.
+                    readFile: function (e) {
+                        const file = e.target.files[0];
+                        if (!file) {
+                            console.log('no file selected');
+                            return;
+                        }
+                        const reader = new FileReader();
+                        reader.onload = this.loadModule;
+                        reader.readAsArrayBuffer(file);
+                    },
+                    // Parse the provided ArrayBuffer as a Wasm module and load it as an Oak module,
+                    // providing the necessary imports.
+                    loadModule: async function (e) {
+                        const contents = e.target.result
+                        console.log('file loaded');
+                        this.module = await WebAssembly.compile(contents);
+                        this.exports = WebAssembly.Module.exports(this.module);
+                        this.imports = WebAssembly.Module.imports(this.module);
+                        // Bind this to a variable so that it can be referred to from the
+                        // implementation of the Oak ABI functions.
+                        const v = this;
```

Comment:
This shouldn't be needed, as arrow functions use a "lexical" this that refers to the outer scope.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

Reply:
Ah thanks, that explains some other weird behaviour I noticed elsewhere when switching from `function` to `=>`, in which `this` points to the wrong thing. I thought the two were equivalent.